tests/tests/tier0/bluechi-change-port-with-c-cmd-option/test_bluechi_change_port_with_c_cmd_option.py

```diff
@@ -4,13 +4,19 @@
 import os
 
 from bluechi_test.test import BluechiTest
-from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine, BluechiMachine
 from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+from bluechi_test.timed_exec import true_within
 from bluechi_test.util import read_file
 
 NODE_FOO = "node-foo"
 
 
+def cmd_on_port(machine: BluechiMachine, port: int, expected_result: int = 0, expected_command: str = None) -> bool:
+    res, output = machine.exec_run(f"bash /var/cmd-on-port.sh {port}")
+    return res == expected_result and expected_command in str(output)
```

Comment:
The code in `bluechi_test.timed_exec` seems a bit too complex. Using the already existing `bluechi_test.util.Timeout` this could be simplified to:

```suggestion
def cmd_on_port(machine: BluechiMachine, port: int, expected_app: str) -> None:
    with Timeout(5, f"Timeout waiting for {expected_app} to listen on port {port}"):
        res, output = machine.exec_run(f"bash /var/cmd-on-port.sh {port}")
        if res == 0 and expected_app in str(output):
            return
        time.sleep(0.2)
```

The error report would also be more clear, I think, since there will be an TimeoutError in the logs with the provided text - compared to an assertion on `True`.